### PR TITLE
Port /reply to the client

### PR DESCRIPTION
**Pull request by bumbadadabum**

### CODE REVIEW

File: js/client-chat.js [hunk 1]
```diff
@@ -857,6 +857,13 @@
 				}
 				return text; // Send the /avatar command through to the server.
 
+			case 'reply':
```
Comment (Zarel) [inline js/client-chat.js]: The replacement should be immediate upon typing `/r ` or `/reply `, it shouldn't wait until you press Enter.

**Comment by bumbadadabum** [comment]
Makes sense, I'll change that.

Op 2 okt. 2016 22:53 schreef "Guangcong Luo" <notifications@github.com>:
>
> @Zarel requested changes on this pull request.
>
> ________________________________
>
> In js/client-chat.js:
>
> > @@ -857,6 +857,13 @@ } return text; // Send the /avatar command through
to the server. + case 'reply':
>
> The replacement should be immediate upon typing /r  or /reply , it
shouldn't wait until you press Enter.

File: js/client-chat.js [hunk 2]
```diff
@@ -148,6 +148,8 @@
 					e.preventDefault();
 					e.stopPropagation();
 				}
+			} else if (app.user.lastPM && textbox.value === '/reply' && e.keyCode === 32) { // '/reply ' is being written
```
Comment (panpawn) [inline js/client-chat.js]: Are we dropping the ``/r`` alias?

Comment (bumbadadabum) [inline js/client-chat.js]: Now we don't anymore.

File: js/client-chat.js [hunk 3]
```diff
@@ -148,6 +148,11 @@
 					e.preventDefault();
 					e.stopPropagation();
 				}
+			} else if (app.user.lastPM && (textbox.value === '/reply' || textbox.value === '/r') && e.keyCode === 32) { // '/reply ' is being written
+				e.preventDefault();
+				e.stopPropagation();
+				var val = textbox.value = '/pm ' + app.user.lastPM + ', ';
```
Comment (Zarel) [inline js/client-chat.js]:
```
var val = '/pm ' + app.user.lastPM + ', ';
textbox.value = val;
```

Don't use fancy tricks on client; you don't know what could cause a deoptimization :|

Comment (kotarou3) [inline js/client-chat.js]: That isn't even a fancy trick. That's like a common thing in basically any imperative language

Comment (Zarel) [inline js/client-chat.js]: UI responsiveness is not IO bound.

This is a common thing in basically any imperative language, yes, but I don't really want to take chances that any engine interprets it as a DOM read.

**Pull request merged**